src/Compilers/CSharp/Test/Emit3/FieldKeywordTests.cs

```diff
@@ -2386,106 +2402,241 @@ public void ReadOnly_02(bool useReadOnlyType, bool useReadOnlyOnGet)
+            string sourceA = $$"""
+                {{typeModifier}} {{typeKind}} S
+                {
+                    {{memberModifier}} object P1 { get; }
+                    {{memberModifier}} object P5 { get; init; }
+                    object P7 { {{memberModifier}} get; init; }
+                    {{memberModifier}} object Q1 { get => field; }
+                    {{memberModifier}} object Q2 { set { _ = field; } }
+                    {{memberModifier}} object Q3 { init { _ = field; } }
+                    {{memberModifier}} object Q4 { get; set { _ = field; } }
+                    {{memberModifier}} object Q5 { get; init { _ = field; } }
+                    object Q6 { {{memberModifier}} get; set { _ = field; } }
+                    object Q7 { {{memberModifier}} get; init { _ = field; } }
+                    object Q8 { get; {{memberModifier}} set { _ = field; } }
+                }
+                """;
+            string sourceB = """
+                using System;
+                using System.Reflection;
+                class Program
                 {
-                    static {{propertyModifier}} object P1 { get; }
-                    static {{propertyModifier}} object P2 { get => field; }
-                    static {{propertyModifier}} object P3 { get => field; set; }
-                    static {{propertyModifier}} object P9 { get; set; }
-                    static {{propertyModifier}} object PA { get; set { } }
-                    static {{propertyModifier}} object PC { get; set { field = value; } }
-                    static {{propertyModifier}} object PD { set { field = value; } }
+                    static void Main()
+                    {
+                        foreach (var field in typeof(S).GetFields(BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static))
+                            Console.WriteLine("{0}: {1}", field.Name, field.IsInitOnly);
+                    }
                 }
                 """;
-            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net80);
-            if (useReadOnlyProperty)
+            var verifier = CompileAndVerify(
+                [sourceA, sourceB],
+                targetFramework: TargetFramework.Net80,
+                verify: Verification.Skipped,
+                expectedOutput: IncludeExpectedOutput($$"""
+                <P1>k__BackingField: True
+                <P5>k__BackingField: True
+                <P7>k__BackingField: True
+                <Q1>k__BackingField: {{useReadOnlyType || useReadOnlyMember}}
+                <Q2>k__BackingField: {{useReadOnlyType || useReadOnlyMember}}
+                <Q3>k__BackingField: True
+                <Q4>k__BackingField: {{useReadOnlyType || useReadOnlyMember}}
+                <Q5>k__BackingField: True
+                <Q6>k__BackingField: {{useReadOnlyType}}
+                <Q7>k__BackingField: True
+                <Q8>k__BackingField: {{useReadOnlyType}}
```

Comment:
It feels like in the case `useReadOnlyType: false, useReadOnlyMember: true`, all accessors are readonly, yet the backing field is not readonly. That seems a little strange. I guess it doesn't really matter, though.

Reply:
Yes, this is an example of the case I was expecting in [https://github.com/dotnet/roslyn/pull/74922#discussion_r1735340269](https://github.com/dotnet/roslyn/pull/74922#discussion_r1735340269).